common/slot_clock/src/lib.rs

```diff
@@ -99,4 +102,14 @@ pub trait SlotClock: Send + Sync + Sized + Clone {
     fn sync_committee_contribution_production_delay(&self) -> Duration {
         self.slot_duration() * 2 / 3
     }
+
+    /// An implementation of the method described in the consensus spec here:
+    ///
+    /// https://github.com/ethereum/consensus-specs/blob/dev/specs/merge/beacon-chain.md#compute_timestamp_at_slot
+    fn compute_timestamp_at_slot(&self, slot: Slot) -> Option<u64> {
```

Comment:
This PR adds a second implementation of `compute_timestamp_at_slot` in the `SlotClock` to avoid accessing the head during block verification just to get the `genesis_time`. The original implementation still exists and is used in the consensus code, because we don't pass the `SlotClock` around the consensus code (and to keep the consensus code more similar to the spec).   Wondering if people think that seems reasonable? Or should we try to keep a single implementation everywhere

Reply:
@paulhauner how does this sound to you?

Reply:
Yep, I think this is a good idea. It's a simple enough function that I think it's worth the trade-off.

Very neat and correct impl too :medal_sports: